src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -283,41 +322,24 @@ private static bool RemovalChangesAssociation(ParenthesizedExpressionSyntax node
                 if (parentExpression is BinaryExpressionSyntax parentBinaryExpression)
                 {
                     // If both the expression and its parent are binary expressions and their kinds
-                    // are the same, check to see if they are commutative (e.g. + or *).
-                    if (parentBinaryExpression.IsKind(SyntaxKind.AddExpression, SyntaxKind.MultiplyExpression) &&
-                        node.Expression.Kind() == parentBinaryExpression.Kind())
+                    // are the same, and the parenthesized expression is on hte right and the 
+                    // operation is associative, it can sometimes be safe to remove these parens.
+                    //
+                    // i.e. if you have "a && (b && c)" it can be converted to "a && b && c" 
+                    // as that new interpretation "(a && b) && c" operates the exact same way at 
+                    // runtime.
+                    //
+                    // Specifically: 
+                    //  1) the operands are still executed in the same order: a, b, then c.
+                    //     So even if they have side effects, it will not matter.
+                    //  2) the same shortcircuiting happens.
+                    //  3) the result will always be the same (for logical operators, there are 
```

Comment:
:bulb: Move 'for logical operators' before the opening parenthesis for improved readability

Reply:
done.